### DISCUSSION TITLE
Zavala 3.0

### DISCUSSION THREAD

**Discussion by vincode-io** [vincode-io/Zavala#220] [General]
I'm currently working towards the 3.0 release of Zavala. Right now I am updating Zavala to work with the latest API's and coding techniques recommend by Apple. This has turned into a huge amount of work and probably won't be completed for some time. I have to touch almost every feature in Zavala in one way or another to get this done.

To make sure I haven't completely broken Zavala with this work, I also plan on writing an automated test suite. That should keep quality up and make regression bugs much less likely to happen. Again, a lot of work that isn't easily noticed in the final product.

The only noticeable features that I plan on doing is enhancing Shortcut support and maybe enabling Markdown imports. For a major release, I like to have at least one new tentpole feature. I haven't decided what that would be. If you have any ideas, comment below and I'll consider them for inclusion in 3.0.

**Comment by azureslumbag** [comment]
I've only just downloaded the app last month, so I apologize if I've missed some progress updates elsewhere in this forum. But Markdown would be helpful since there's so many other programs use Markdown to organize text. But one thing I noticed in Obsidian, which I gave up on, was that Markdown and bullet points didn't play so well. I had huge issues indenting, bulleting (1, 2, 3), and assigning headers. Zavala is super clean right now without Markdown!

For the shortcuts, would Zavala be the starting point or end point? I'm not a programmer or coder, and I have issues with shortcuts - I can't even get a shortcut to save a currently playing song to a playlist. 

Are there any plans to introduce a legal numbering system? Not sure how much of a PITA that is, but it would play well with project workflows if that was a feature! Thanks for making a really easy to use, very clear app. It's helped me this year.

**Comment by vincode-io** [comment]
@azureslumbag We don't have any plans to support Markdown syntax inside Zavala. Zavala is a rich text outliner that will export to Markdown and possibly import Markdown in the future.

Zavala can be the source or receiver of data when using Shortcuts.

We may introduce a numbering system at some point. Right now it isn't a super high priority for the project.
